secret_store/src/key_server_cluster/math.rs

```diff
@@ -418,10 +429,108 @@ pub fn verify_signature(public: &Public, signature: &(Secret, Secret), message_h
 	Ok(combined_hash == signature.0)
 }
 
+
+/// Compute R part of ECDSA signature.
+#[cfg(test)]
+pub fn compute_ecdsa_r(nonce_public: &Public) -> Result<Secret, Error> {
+	to_scalar(nonce_public[0..32].into())
+}
+
+/// Compute share of S part of ECDSA signature.
+#[cfg(test)]
+pub fn compute_ecdsa_s_share(inv_nonce_share: &Secret, inv_nonce_mul_secret: &Secret, signature_r: &Secret, message_hash: &Secret) -> Result<Secret, Error> {
+	let mut nonce_inv_share_mul_message_hash = inv_nonce_share.clone();
+	nonce_inv_share_mul_message_hash.mul(&message_hash.clone().into())?;
+
+	let mut nonce_inv_share_mul_secret_share_mul_r = inv_nonce_mul_secret.clone();
+	nonce_inv_share_mul_secret_share_mul_r.mul(signature_r)?;
+
+	let mut signature_s_share = nonce_inv_share_mul_message_hash;
+	signature_s_share.add(&nonce_inv_share_mul_secret_share_mul_r)?;
+
+	Ok(signature_s_share)
+}
+
+/// Compute S part of ECDSA signature from shares.
+#[cfg(test)]
+pub fn compute_ecdsa_s(t: usize, signature_s_shares: &[Secret], id_numbers: &[Secret]) -> Result<Secret, Error> {
+	let double_t = t * 2;
+	debug_assert!(id_numbers.len() >= double_t + 1);
+	debug_assert_eq!(signature_s_shares.len(), id_numbers.len());
+
+	compute_joint_secret_from_shares(double_t,
```

Comment:
This interpolation doesn't seem to be possible with only `t` participants (as it's polluted by the shares of the `z` blinding factor).

Reply:
yes, but if running `compute_ecdsa_s` is part of a t-of-n scheme then I don't understand how it could ever complete (relevant to my other comment)

Reply:
or have I got it backwards somehow, and it's actually a 2t-of-n scheme because we have to interpolate such polynomials?

Reply:
`signature_s_shares` are the shares of S in `(2*t+1)-of-N` scheme. So what `compute_ecdsa_s` does is exactly interpolating 2*t-degree polynomial.